src/main/java/org/broadinstitute/barclay/argparser/Argument.java

```diff
@@ -0,0 +1,106 @@
+ * would have annotations on fields to handle the values of option and x. The java type of the option
+ * will be inferred from the type of the field or from the generic type of the collection
+ * if this option is allowed more than once. The type must be an enum or
+ * have a constructor with a single String parameter.
+ */
+@Retention(RetentionPolicy.RUNTIME)
+@Target(ElementType.FIELD)
+@Documented
+public @interface Argument {
+
+    /**
+     * The full name of the command-line argument.  Full names should be
+     * prefixed on the command-line with a double dash (--).
+     * @return Selected full name, or "" to use the default.
+     */
+    String fullName() default "";
+
+    /**
+     * Specified short name of the command.  Short names should be prefixed
+     * with a single dash.  Argument values can directly abut single-char
+     * short names or be separated from them by a space.
+     * @return Selected short name, or "" for none.
+     */
+    String shortName() default "";
+
+    /**
+     * Documentation for the command-line argument.  Should appear when the
+     * --help argument is specified.
+     * @return Doc string associated with this command-line argument.
+     */
+    String doc() default "Undocumented option";
+
+    /**
+     * If set to false, a {@link org.broadinstitute.barclay.argparser.CommandLineException.MissingArgument} will be thrown
+     * if the option is not specified.
+     * If 2 options are mutually exclusive and both are required it will be interpreted as one or the other is required
+     * and an exception will only be thrown if neither are specified.
+     * An argument with a non-null default value specified will ignore this flag and always be treated as optional
+     */
+    boolean optional() default false;
+
+    /**
+     * Array of option names that cannot be used in conjunction with this one.
+     * If 2 options are mutually exclusive and both have optional=false it will be
+     * interpreted as one OR the other is required and an exception will only be thrown if
+     * neither are specified.
+     */
+    String[] mutex() default {};
+
+    /**
+     * Is this an Option common to all command line programs.  If it is then it will only
+     * be displayed in usage info when H or STDHELP is used to display usage.
+     */
+    boolean common() default false;
+
+    /**
+     * Does this option have special treatment in the argument parsing system.
+     * Some examples are arguments_file and help, which have special behavior in the parser.
+     * This is intended for documenting these options.
+     */
+    boolean special() default false;
+
+    /**
+     * Are the contents of this argument private and should be kept out of logs.
+     * Examples of sensitive arguments are encryption and api keys.
+     */
+    boolean sensitive() default false;
+
+    //TODO: Start used by the legacy command line parser only.
```

Comment:
This was a note to myself to keep track of stuff that was legacy only. Will remove it now.